.github/workflows/prerelease.yml

```diff
@@ -7,208 +7,14 @@ on:
-        if: startsWith(matrix.test-upgrade, 'false')
-        shell: bash
-        run: |
-          build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
-      - name: Notify failure via Slack
-        if: ${{ failure() }}
-        uses: archive/github-actions-slack@master
-        with:
-          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
-          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
-          slack-text: "❌ `${{ env.REPO_FULL_NAME }}`: prerelease pipeline failed."
-
-  publish-to-s3:
-    name: Send release assets to S3
-    runs-on: ubuntu-20.04
-    needs: [package-win]
-    steps:
-      - name: Login to DockerHub
-        uses: docker/login-action@v3
-        with:
-          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
-          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
-      - name: Publish to S3 action
-        uses: newrelic/infrastructure-publish-action@v1
-        env:
-          AWS_S3_BUCKET_NAME: "nr-downloads-ohai-staging"
-          AWS_S3_LOCK_BUCKET_NAME: "onhost-ci-lock-staging"
-        with:
-          disable_lock: false
-          run_id: ${{ github.run_id }}
-          tag: ${{env.TAG}}
-          app_name: "nri-${{env.INTEGRATION}}"
-          repo_name: ${{ env.ORIGINAL_REPO_NAME }}
-          access_point_host: "staging"
-          schema: "ohi-jmx" # ohi-jmx for integrations that bundle JMX on windows installers
-          aws_region: "us-east-1"
-          aws_role_arn: ${{ secrets.OHAI_AWS_ROLE_ARN_STAGING }}
-          aws_role_session_name: ${{ secrets.OHAI_AWS_ROLE_SESSION_NAME_STAGING }}
-          aws_access_key_id: ${{ secrets.OHAI_AWS_ACCESS_KEY_ID_STAGING }}
-          aws_secret_access_key: ${{ secrets.OHAI_AWS_SECRET_ACCESS_KEY_STAGING }}
-          aws_s3_bucket_name: ${{ env.AWS_S3_BUCKET_NAME }}
-          # used for locking in case of concurrent releases
-          aws_s3_lock_bucket_name: ${{ env.AWS_S3_LOCK_BUCKET_NAME }}
-          # used for signing package stuff
-          gpg_passphrase: ${{ secrets.OHAI_GPG_PASSPHRASE }}
-          gpg_private_key_base64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }}
-      - name: Test uploaded package installability
-        uses: newrelic/integrations-pkg-test-action/linux@v1
-        with:
-          tag: ${{ env.TAG }}
-          integration: nri-${{ env.INTEGRATION }}
-          packageLocation: repo
-          stagingRepo: true
-          upgrade: false
+  pre-release:
+    uses: newrelic/coreint-automation/.github/workflows/reusable_pre_release.yaml@v3
+    with:
+      tag: ${{ github.event.release.tag_name }}
+      integration: "jmx"
+      windows_goarch_matrix: ["amd64"] # 386 not supported in jmx integrations
+      win_package_type: exe # JMX integrations are shipped in .exe format
+      publish_schema: "ohi-jmx" # ohi-jmx for integrations that bundle JMX on windows installers
+      NRJMX_VERSION: '2.6.0' # TODO check if we should update it to latest release version
```

Comment:
2.6.0 is the latest, isn't it? Would it make sense having a custom rule to make renovate updating this dependency?

Reply:
I see [v3.7.1](https://github.com/newrelic/nri-jmx/releases/tag/v3.7.1#:~:text=v3.7.1-,v3.7.1,-Latest) as the latest version in the releases tab

Reply:
That link is for this repo, `nri-jmx`,  but it is pointing to `nrjmx`, isn't it? https://github.com/newrelic/nrjmx/releases

Reply:
yeah got confused by the naming :(
Did we have a custom rule in renovate for any other repo?

Reply:
We have some custom rules in the prometheus-exporter-packages repository: https://github.com/newrelic/newrelic-prometheus-exporters-packages/blob/main/.github/renovate.json5

Reply:
Should we be automerging the dependency bump for all new major, minor, patch version releases?

Reply:
Even if an _internal_ dependency, I would not automerge the major version (after all, major means breaking changes). I'd say it would be nice to merge minors and patches automatically and get a PR for majors.

Reply:
I have added the renovate config to detect new versions of nrjmx and auto upgrade the value of NRJMX_VERSION in `.github/workflows/push_pr.yml`, `.github/workflows/prerelease.yml` files. Tested it in the forked repo and it has automerged the [PR](https://github.com/sairaj18/nri-jmx-test/pull/9).

Reply:
Thanks!